presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -157,7 +157,7 @@ public StatisticRange addAndSumDistinctValues(StatisticRange other)
 
     public StatisticRange addAndMaxDistinctValues(StatisticRange other)
     {
-        double newDistinctValues = minExcludeNaN(distinctValues, other.distinctValues);
+        double newDistinctValues = max(distinctValues, other.distinctValues);
```

Comment:
Commit message says about ExchangeStatsRule. Is this method used from that rule only?

Reply:
Yeah. I'm working on adding more comprehensive unit tests for the `PlanNodeStatsEstimateMath` as part of the https://github.com/prestodb/presto/pull/11973